msal/application.py

```diff
@@ -194,12 +194,16 @@ def __init__(
                 self.http_client.request, timeout=timeout)
         self.app_name = app_name
         self.app_version = app_version
+        self.authority_url = authority
+        self.validate_authority = validate_authority
         self.authority = Authority(
```

Comment:
Since we are heading to this direction anyway, let's do it thoroughly, by also lazy-building the Authority instance. One of the approaches to do this, is to add something like this at each API which needs to use `self.authority`:

```python
if not self.authority.is_initialized:
    self.authority.initialize()  # It will send out network calls which is currently inside Authority constructor
```